posts/google-summer-of-code-fellows/gsoc-2024-fellows.qmd

```diff
@@ -0,0 +1,84 @@
+
+2. Find Treatment History of Patients:  Create the treatment history of a patient based on target, event, and exit cohorts. Then filter patient events based on the start and end dates of the target cohort. Third, Calculate the duration of treatment eras and gap between treatments.
+
+3. Filters: Filter the treatment history based on the minEraDuration parameter and EraCollapse parameter.
+
+4. Create Continuous integration and Continuous Development pipeline for the package. 
+
+// Stretch Goals:
+
+(a) Composing with JuliaStats Ecosystem
+(b) Novel Visualizations for Pathways
+
+
+# Work Details
+
+**1. Setting up the package in JuliaHealth Channel**
+
+Initially, there was no package as such for pathways synthesis, so I had to build it from scratch. Firstly, the repository by the name [**OMOPCDMPathways.jl**](https://github.com/JuliaHealth/OMOPCDMPathways.jl). Once the repository was created, we were needed to have a skeleton for an standard Julia repository. For this we used the [PkgTemplate.jl](https://juliaci.github.io/PkgTemplates.jl/stable/user/) this provided a basic skeleton for the repository that included - folders for test suites, documentations, src [code files], GitHub files, Readme and Licensing file, toml and citation files. All this we can further edit and modify as per our work. By default PkgTemplate.jl uses [Documenter.jl](https://documenter.juliadocs.org/stable/) for the documentations part but as suggested and discussed with my mentor we decided to shift to [DocumenterVitepress.jl](https://luxdl.github.io/DocumenterVitepress.jl/dev/) for the documentations part.After we had shifted to DocumenterVitepress the main task now was to host the documentations, this was done using Github-Actions, detailed steps for hosting is provided at [this](https://documenter.juliadocs.org/stable/man/hosting/#Hosting-Documentation) page.Then we added the codecov to our package by trigering it via a dummy function and a corresponding test case for it. Also with the CI for the package was set up with it.And, now finally the repository was ready with test coverage, CI, documentations fully functionaly repository ready.
+
+
+**2. Loading the PostgreSQL Database**
+
+The main database we worked / has build analysis was on the freely avaiable OMOPCDM Database. 
+However, I was provided with some more extra synthetic databse from my mentor for further testing of the functionalities. 
+Being a very large database I had to strategcly download it futher, my mentor helped me in setting up the postgres on my local machine.
+Once, the database was set up proper testing was performed on it to check if things are as expected. And with this we were done with the database set up as well
+and could finally dive into the actual code logic for the Pathways synthesis.
+
+
+**3. Selecting treatments of interest**
+
+So, as a part of this we used the previously mentioned research paper and discussion with the mentors we came up with logic for it.
```

Comment:
Oh could you actually include the paper in the background as well as the R reference?

Reply:
Yeah, the R paper I can include in the patient pathways section